### PR TITLE
PublicDashboards: Sanitize metadata from public dashboard queries

### PR DESCRIPTION
**Pull request by guicaulada**
**What this PR does / why we need it**:

This PR adds a function to remove metadata from queries on public dashboards.

This is needed because metadata on query responses can contain sensitive information that is used for diagnostics purposes and shouldn't be available for open access.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/4453

**Special notes for your reviewer**:

I'm not sure if the place where I added this function is the best place to have it. I think the "remove metadata" issue is very much related only to Public Dashboards, but I can see how delegating this reponsability to the QueryDataService could make sense.

### CODE REVIEW

File: pkg/services/publicdashboards/queries/dashboard_queries.go [hunk 1]
```diff
@@ -100,3 +101,12 @@ func GetDataSourceUidFromJson(query *simplejson.Json) string {
 
 	return uid
 }
+
+func RemoveMetadataFromQueryData(res *backend.QueryDataResponse) {
```
Comment (owensmallwood) [inline pkg/services/publicdashboards/queries/dashboard_queries.go:105]: We should probably have a unit test for this

Comment (guicaulada) [reply pkg/services/publicdashboards/queries/dashboard_queries.go:105]: Yup! I was planning on adding that after the approach was confirmed 👍

Comment (guicaulada) [reply pkg/services/publicdashboards/queries/dashboard_queries.go:105]: Done on https://github.com/grafana/grafana/pull/55269/commits/8b6fedf643290b0b08ad02274279349af962f59f

**Review by owensmallwood — APPROVED** [review]
Looks good! 👍

**Comment by guicaulada** [comment]
Sure! I will update only to remove those specific values!

I thought it would be best to iteratively update the code since I wasn't sure which values can be removed.

Do you know which values can be safely removed and contain sensitive information?

**Comment by guicaulada** [comment]
@yesoreyeram done on https://github.com/grafana/grafana/pull/55269/commits/df0c6cde56a9d1d91eddbc3255cb852c91f05958

It's easy enough to remove new fields in the future if we see they are not being used and contain sensitive information.

**Pull request merged**